src/Orleans.Runtime/Hosting/DefaultSiloServices.cs

```diff
@@ -130,7 +128,7 @@ internal static void AddDefaultServices(HostBuilderContext context, IServiceColl
             services.TryAddSingleton<InsideRuntimeClient>();
             services.TryAddFromExisting<IRuntimeClient, InsideRuntimeClient>();
             services.TryAddFromExisting<ISiloRuntimeClient, InsideRuntimeClient>();
-            services.TryAddFromExisting<ILifecycleParticipant<ISiloLifecycle>, InsideRuntimeClient>();
+            services.AddFromExisting<ILifecycleParticipant<ISiloLifecycle>, InsideRuntimeClient>();
```

Comment:
Why is this AddFromExisting, not TryAddFromExisting. We tend to follow that pattern in this file

Reply:
Because Try here will find other registrations of `ILifecycleParticipant<ISiloLifecycle>` and not add this one. See https://github.com/dotnet/orleans/pull/3738#discussion_r155381597